Отрисуй меня полностью (часть 1)

ERS86 opened the pull request:
Если не считать подключения main.js, то это даже не попытка, а просто желание показать ход мыслей. С надеждой получить необходимые подсказки.

---
:mortar_board: [Отрисуй меня полностью (часть 1)](https://up.htmlacademy.ru/javascript/26/user/2042857/tasks/12)

:boom: https://htmlacademy-javascript.github.io/2042857-kekstagram-26/13/

ERS86 committed:
Первая попытка

Если не считать подключения main.js, то это даже не попытка, а просто желание показать ход мыслей. С надеждой получить необходимые подсказки.

NightDiRaven commented on js/data.js:
```diff
@@ -19,14 +19,14 @@ const COMMENT_MESSAGES = [
 const photoDescriptions = [];
 const idsForComments = [];
-const MAX_COMMENTS_COUNT = 10;
+export const MAX_COMMENTS_COUNT = 10;
```
нет необходимости экспортировать

NightDiRaven commented on js/data.js:
```diff
@@ -19,14 +19,14 @@ const COMMENT_MESSAGES = [
 }
 
-const getPhotoUrl = function () {
+export const getPhotoUrl = function () {
```
нет необходимости экспортировать

NightDiRaven commented on js/data.js:
```diff
@@ -59,7 +59,7 @@ const getComments = function (count) {
 };
 
-const getNewPhoto = (id) => ({
+export const getNewPhoto = (id) => ({
```
Тоже, зачем? Можно, но не нужно

NightDiRaven replied:
нужно экспортировать только createDescriptions и MAX_DESCRIPTION_COUNT

NightDiRaven replied:
Создание описаний `createDescriptions(MAX_DESCRIPTION_COUNT);` отсюда убираешь и переносишь в main.js

NightDiRaven commented on js/main.js:
```diff
@@ -1 +1,2 @@
 import './data.js';
```
```suggestion
import {createDescriptions, MAX_DESCRIPTION_COUNT} from './data.js'; // импортируешь функцию создания описаний (фотографий)
import {drawPictures} from './miniatures.js; // импортируешь функцию "отрисовки" данных фотографий в дом елементы


const descriptions = createDescriptions(MAX_DESCRIPTION_COUNT);  // Создаешь массив фотографий с описаниями и комментами

drawMiniatures(descriptions) // вызываешь отрисовку всех картинок
```

NightDiRaven commented on js/miniautures.js:
```diff
@@ -0,0 +1,31 @@
+}
+
+pictures.append(fragment);
+
```
Код отрисовки примерно верный, но идея не совсем верна: нужно один раз сгененрировать данные, а потом их уже использовать и отрисовывать именно их, предлагаю, например, такой вариант, вместо твоего

```suggestion
const drawPicture = funcition (description) {
    const newPhoto = template.cloneNode(true); // Клонируем элемент со всеми "внутренностями"
    
    // Заполняем нашими данными
    newPhoto.img.src = description.url ; 
    newPhoto.pictures__likes = description.likes;
    newPhoto.pictures__comments = description.comments.length;
    
    return newPhoto;
}

export const drawMiniatures = funcition (descriptions) {
const fragment = document.createDocumentFragment();
  descriptions.forEach(description => {
    const pictureEl = drawPicture(description) // для каждого описания создаешь изображение
       fragment.appendChild(pictureEl );
  })
 pictures.append(fragment);
} 
```

NightDiRaven reviewed (changes requested):
Код отрисовки примерно верный, но идея не совсем верна: нужно один раз сгененрировать данные, а потом их уже использовать

ERS86 committed:
Отрисуй меня полностью (часть 1) #2

Не уверен, что значения передаются именно таким путём, через точки.
newPhoto.img.src = photo.url

ERS86 committed:
Отрисуй меня полностью (часть 1), попытка 3

ERS86 committed:
Исправления после check

ERS86 committed:
Исправление после check_2

NightDiRaven commented on index.html:
```diff
@@ -232,7 +232,7 @@ <h2 class="success__title">Изображение успешно загруже
   </template>
 
-  <script src='js/main.js'></script>
+  <script src='/js/main.js'></script>
```
Если работаешь с модулями (а так нужно, это export/import) то нужно явно указать  это при подключении файла
```suggestion
  <script src='/js/main.js' type="module"></script>
```

NightDiRaven commented on js/miniautures.js:
```diff
@@ -0,0 +1,28 @@
+import  {urlForExport, likesForExport, commentsForExport} from './data.js';
```
Это не нужно, ты будешь получать данные в качестве аргументов функции, а не так, данные у тебя уже есть, ты их передал в функцию drawPictures, убери

NightDiRaven commented on js/miniautures.js:
```diff
@@ -0,0 +1,28 @@
+  .content
+  .querySelector('a');
+
+const drawPicture = function () {
```
Добавь аргумент (функция будет принимать данные одной фотографии)
```suggestion
const drawPicture = function (picture) {
```

NightDiRaven commented on js/miniautures.js:
```diff
@@ -0,0 +1,28 @@
+const drawPicture = function () {
+  const newPicture = pictureTemplate.cloneNode(true);
+
+  const pictureSrcElement = newPicture.querySelector('src');
```
Не тот селектор
```suggestion
  const pictureSrcElement = newPicture.querySelector('.picture__img');
```

NightDiRaven commented on js/miniautures.js:
```diff
@@ -0,0 +1,28 @@
+  const newPicture = pictureTemplate.cloneNode(true);
+
+  const pictureSrcElement = newPicture.querySelector('src');
+  pictureSrcElement.textContent = urlForExport;
```
у объекта ноды изображение свойство src в него записываем урл нашей картинки, которую получили в качестве аргумента функции
```suggestion
 pictureSrcElement.src = picture.url;
```

NightDiRaven commented on js/miniautures.js:
```diff
@@ -0,0 +1,28 @@
+  const pictureSrcElement = newPicture.querySelector('src');
+  pictureSrcElement.textContent = urlForExport;
+  const pictureLikesElement = newPicture.querySelector('.picture__likes');
+  pictureLikesElement.textContent = likesForExport;
```
Текстовое содержимое элемента с количеством лайков устанавливаем как количество лайков нашей картинки, которую получили в качестве аргумента функции
```suggestion
pictureLikesElement.textContent = picture.likes;
```

NightDiRaven commented on js/miniautures.js:
```diff
@@ -0,0 +1,28 @@
+  const pictureLikesElement = newPicture.querySelector('.picture__likes');
+  pictureLikesElement.textContent = likesForExport;
+  const pictureCommentsElement = newPicture.querySelector('.picture__comments');
+  pictureCommentsElement.textContent = commentsForExport;
```
Текстовое содержимое элемента с количеством комментов устанавливаем как длина массива комментариев нашей картинки, которую получили в качестве аргумента функции (это и есть колчество комментов)
```suggestion
pictureCommentsElement.textContent = picture.comments.length;
```

ERS86 committed:
Отрисуй меня полностью (часть 1)

Очередная проверка

ERS86 committed:
Отрисуй меня полностью (часть1)2
